### DISCUSSION TITLE
Add modifier property

### DISCUSSION THREAD

**Discussion by dczuo** [dotnet/csharplang#6430] [Language Ideas]
Add modifier property to simplify property declaration
```csharp
public partial class User
{
    property string Name;  // equivalent to public partial string Name { get; set; }
}
```

**Comment by CyrusNajmabadi** [comment]
I don't think we would ever default properties to be read/write. If we had this, it would be a get-only property, or get/init.

**Comment by HaloFour** [comment]
The default accessibility of any member is `private`, not `public`, so I would not expect that a `property` keyword would have a different default accessibility.

Also, why would it be `partial` by default?  That only applies to very specific use cases that would likely involve source generators to provide the full implementation.

**Comment by TahirAhmadov** [comment]
I'm not sure adding a new keyword is worth shifting a few keystrokes here, to be honest.

**Comment by bernd5** [comment]
It would be similar to event-field declarations - but then I would allow the keyword (like events) for explicit implemented properties, too.
Accessibility / partial should not implied by `property`.
